pkgs/servers/dnschain/default.nix

```diff
@@ -1,11 +1,51 @@
-{ stdenv, nodePackages }:
-
-# to update dnschain, run npm2nix package.json package.nix, and
-# then add "coffee-script" manually as a dependecy for "dnschain"
-# in package.nix.
+{ stdenv, fetchFromGitHub, callPackage, makeWrapper, openssl }:
 
 let
-  np = nodePackages.override { generated = ./package.nix; self = np; };
-in
+  nodePackages = callPackage (import ../../top-level/node-packages.nix) {
```

Comment:
Why did you replace the previous expression by this import statement ? They are basically the same, but yours takes more space and uses top-level/node-packages.nix directly. Why not use nodePackages.override ?